python/lsst/sconsUtils/tests.py

```diff
@@ -293,3 +305,38 @@ def junitPrefix(self):
             prefix += ".{0}".format(os.environ[controlVar])
 
         return prefix
+
+    def _pytestCoverage(self, prefix="pytest"):
+        """Form the additional arguments required to enable coverage testing.
+        Prefix is used to form the output files.
```

Comment:
I think this docstring needs to breakdown what the various possibilities for `prefix` do. I'm having trouble parsing it from the code itself.

If the goal is to just turn on or off html and xml output, why not just have explicit options for those (`terminal=True, html=True, xml=False` seems like a sensible default), and then Jenkins or whatever can configure that as part of `run()`.

Reply:
The prefix really is the prefix to use for the output files, since each pytest invocation needs to have different output files. I think I can remove prefix if I use Scons `${TARGET}` instead. If I do that I solve your problem of writing the files in `.tests` but they get directory names like `tests/.tests/pytest-utils.xml-htmlcov` (for utils). If people don't mind the extra `.xml` in there then the code is much much simpler.